src/app/clusters/general-commissioning-server/general-commissioning-server.cpp

```diff
@@ -127,13 +127,27 @@ bool emberAfGeneralCommissioningClusterArmFailSafeCallback(app::CommandHandler *
                                                            const app::ConcreteCommandPath & commandPath,
                                                            const Commands::ArmFailSafe::DecodableType & commandData)
 {
-    DeviceControlServer * server = &DeviceLayer::DeviceControlServer::DeviceControlSvr();
-    CheckSuccess(server->ArmFailSafe(System::Clock::Seconds16(commandData.expiryLengthSeconds)), Failure);
-
+    FailSafeContext & failSafeContext = DeviceLayer::DeviceControlServer::DeviceControlSvr().GetFailSafeContext();
     Commands::ArmFailSafeResponse::Type response;
-    response.errorCode = CommissioningError::kOk;
-    response.debugText = CharSpan("", 0);
-    CheckSuccess(commandObj->AddResponseData(commandPath, response), Failure);
+
+    /*
+     * If the fail-safe timer was not currently armed, the the fail-safe timer SHALL be armed.
+     * If the fail-safe timer was currently armed, and current accessing fabric matches the fail-safe
+     * context’s Fabric Index, then the fail-safe timer SHALL be re-armed.
+     */
+    if (!failSafeContext.IsFailSafeArmed())
+    {
+        CheckSuccess(failSafeContext.ArmFailSafe(commandObj->GetAccessingFabricIndex(),
+                                                 System::Clock::Seconds16(commandData.expiryLengthSeconds)),
+                     Failure);
+        response.errorCode = CommissioningError::kOk;
+        CheckSuccess(commandObj->AddResponseData(commandPath, response), Failure);
+    }
+    else
+    {
+        response.errorCode = CommissioningError::kBusyWithOtherAdmin;
```

Comment:
This code very much does not match the comment (or the spec, which the comment matches).

Reply:
Why it is not match? the accessing fabric matches now is done within ArmFailSafe

Reply:
If the failsafe is already armed, this code always errors out.  The spec does not error out if it's armed for the accessing fabric index.

Reply:
Move the  accessing fabric matches out of ArmFailSafe and do the check at the callers